File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/Tablero.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author pc
 */
public class Tablero
{
    private int numCasillaCarcel;
    private ArrayList <Casilla> casillas;
    private int porSalida;
    private boolean tieneJuez;
    
    Tablero (int pos_carcel)
    {
        if (pos_carcel >=1)
            numCasillaCarcel = pos_carcel;
        else
            numCasillaCarcel = 1;
        
        casillas=new ArrayList();
          
        Casilla salida = new Casilla ("Salida");
        casillas.add(salida);
        porSalida=0;
        tieneJuez=false;
        
    }
    
    
    private Boolean correcto()
    {
        Boolean es_correcto = false;
        
        if ((casillas.size() > numCasillaCarcel)&&tieneJuez)
        {
            es_correcto = true;
        }
        return es_correcto;
    }
    private Boolean correcto(int numCasilla)
    {
        Boolean es_correcto = false;
        
        if (this.correcto() && numCasilla <= casillas.size())
        {
            es_correcto = true;
        }
        return es_correcto;
    }
    
    int getCarcel (){
        return numCasillaCarcel;
    }
    
    int getPorSalida()
    {
        int aux = porSalida;
        if (porSalida > 0)
        {
            porSalida = porSalida - 1;
        }    
        return aux;
    }
    
    void añadeCasilla (Casilla casilla)
    {
        if (casillas.size()== numCasillaCarcel)
        {
            Casilla carcel = new Casilla ("Cárcel");
            casillas.add(carcel);
        }
        casillas.add(casilla);
        
    }
    void añadeJuez()
    {
        if (tieneJuez==false){
            Casilla juez = new Casilla ("Juez");
            casillas.add(juez);
            tieneJuez=true;
        }
    }
    Casilla getCasilla (int numCasilla)
    {
        Casilla casilla = null;
        if (correcto(numCasilla))
        {
            casilla = casillas.get(numCasilla);
        }
        return casilla;
    }
    int nuevaPosicion (int actual, int tirada)
    {
        if(correcto()){
            int pos1=actual+tirada;
            int pos2=(pos1)%casillas.size();
            if(pos1==pos2)
                return pos1;
            else 
            {
                porSalida++;
                return pos2;
            }
        }
        else
            return -1;
    }
    
    int calcularTirada (int origen, int destino)
    {
        int tirada;
        tirada = destino - origen;
        if (tirada <0){
            porSalida++;
            tirada = tirada + casillas.size();
        }
        return tirada;
    }
}

File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/tablero.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
  

class Tablero
  require_relative ('casilla')
  attr_reader :numCasillaCarcel
    def initialize(indice)
      if(indice>=1)
        @numCasillaCarcel=indice
      else
        @numCasillaCarcel=1  
      
      end
      @casillas=Array.new
      @casillas<<Casilla.new("Salida")
 
      @porSalida=0
      @tieneJuez=false
    end
    
#    private
#    def tablero_correcto
#      
#      es_correcto = false
#      if @casillas.size > @numCasillaCarcel then
#      es_correcto = true  
#      end
#      es_correcto
#    end
    private
    def correcto(numCasilla=nil)
      if(numCasilla==nil)
       return @tieneJuez && @casillas.size > @numCasillaCarcel
      else
        
       if(@tieneJuez && @casillas.size > @numCasillaCarcel)
            return @casillas.size > numCasilla
          end   
      end
    end
 
    public
    def getPorSalida
      salida=@porSalida
      if @porSalida>0 then
    
      @porSalida=@porSalida - 1
      
      end
      
      salida
    end
    
    def aniadeCasilla (casilla)
      
      if(@casillas.size==@numCasillaCarcel)
        @casillas<<Casilla.new("Cárcel")
        @casillas<<casilla
      else
        @casillas<<casilla
      end
    end
    
    
    def aniadeJuez
      if(!@tieneJuez)
        @casillas<<CasillaJuez.new("Juez",@numCasillaCarcel)
        @tieneJuez=true
      end
      
    end
    
    
    def getCasilla (numCasilla)
      casilla=nil
      if numCasilla <= @casillas.size then
        casilla = @casillas.at(numCasilla)
      end
      return casilla
    end
    
    def nuevaPosicion (actual,tirada) 
      if correcto
        pos = actual + tirada
        pos = pos % @casillas.size
        if(pos==actual + tirada)
          return actual + tirada
        else
          @poSalida=@porSalida+1
          return pos
        end
      else
        return -1
      end
    end  
    
    
    
    def calcularTirada (origen,destino)
   
        tirada = destino - origen;
        if tirada < 0 then
          @porSalida=@porSalida+1
          tirada = tirada + @casillas.size
        end
        return tirada
    end
    
end  
end    
    
 